src/Xamarin.Android.Tools.AndroidSdk/Sdks/AndroidSdkWindows.cs

```diff
@@ -193,6 +191,51 @@ private static IEnumerable<string> GetKnownOpenJdkPaths ()
 				.Select (openJdk => openJdk.Item1);
 		}
 
+		static IEnumerable<string> GetMicrosoftOpenJdkRegistryPaths ()
+		{
+			var paths   = new List<(Version version, string path)> ();
+			var roots   = new[] { RegistryEx.CurrentUser, RegistryEx.LocalMachine };
+			var wows    = new[] { RegistryEx.Wow64.Key32, RegistryEx.Wow64.Key64 };
+			foreach (var root in roots)
+			foreach (var wow in wows) {
+				foreach (var subkeyName in RegistryEx.EnumerateSubkeys (root, MICROSOFT_OPENJDK_PATH, wow)) {
+					if (!Version.TryParse (subkeyName, out var version))
+						continue;
+					var msiKey  = $@"{MICROSOFT_OPENJDK_PATH}\{subkeyName}\hotspot\MSI";
+					var path    = RegistryEx.GetValueString (root, msiKey, "Path", wow);
+					if (path == null)
+						continue;
+					paths.Add ((version, path));
+				}
+			}
+
+			return paths.OrderByDescending (e => e.version)
+				.Select (e => e.path);
+		}
+
+		static Version? ExtractVersion (string path, string prefix)
```

Comment:
Since this method does a bit of string manipulation, would it be worth writing a few unit tests that call it? To check you get the right result.

Reply:
Good call; found at least one infinite loop doing that!